web/packages/shared/utils/highbar.ts

```diff
@@ -60,33 +60,55 @@ export function mergeDeep(target: MergeTarget, ...sources: Array<MergeTarget>) {
   return mergeDeep(target, ...sources);
 }
 
-type CompareArray = Array<Record<string, unknown>>;
+/** Recursively freezes an object and its contents. */
```

Comment:
Why is the deep freezing is necessary for the role editor?

Reply:
It was just a precaution, since I declared a default options object as a constant, against accidental modifying a constant. However, as I wrote, I decided to just create each instance in place.